index.js

```diff
@@ -30,7 +35,16 @@ const streamFeaturesFromFile = (filePath) => {
       ) {
         return streamFromMapnik(featureStream, filePath, fileType);
       }
-      throw invalid(`Unknown file type "${fileType}": accepts .geojson, .csv, or .shp`);
+      if (fileType === 'zip') {
+        shpFairy(filePath, (err, shpFile) => {
+          if (err && err.code === 'EINVALID') {
+            return handleError(invalid('Invalid zipfile: ' + err.message));
+          }
+          if (err) return handleError(err);
+          return streamFromMapnik(featureStream, shpFile, 'shp');
+        });
+      }
+      else return handleError(invalid(`Unknown file type "${fileType}": accepts .geojson, .csv, or .shp (zipped and unzipped)`));
     })
     .catch((err) => featureStream.emit('error', err));
```

Comment:
This catch statement is going to grab any errors that are thrown or returned Promises that are rejected in the chain. So probably instead of handling errors with two different functions, we want to make any `shpFairy` errors fit this existing pattern. I think one way you could do this would be to create a `streamFromZip` module that returns a Promise, as the `streamFromGeoJson` and `streamFromMapnik` functions do.

`streamFromZip` could use this handy pattern for wrapping callback-style functions in a Promise:

```js
function streamFromZip(featureStream, filePath) {
  return Promise((resolve, reject) => {
    shpFairy(filePath, (err, shpFile) => {
      if (err && err.code === 'EINVALID') return reject(invalid(`Invalid zipfile: ${err.message}`));
      if (err) return reject(err);
      return resolve(streamFromMapnik(featureStream, shpFile, 'shp'));
    });
  });
}
```

Then in `index.js` you could:

```js
if (fileType === 'zip') {
  return streamFromZip(featureStream, filePath);
}
```

*But also* for the sake of answering your direct question, I think your error would be solved by adding braces separating `else` from the `return` statement:

```js
} else {
  return handleError(..);
}
```

ESLint just wants `}` and `else` to be on the same line.